test/mock/createInsightsClient.ts

```diff
@@ -1,85 +1,89 @@
-    setValues({ _hasCredentials: true, _appId: appId, _apiKey: apiKey });
-    setUserToken(ANONYMOUS_TOKEN);
-  };
-  const _get = (key, callback) => callback(values[key]);
-  const onUserTokenChange = (callback, { immediate = false } = {}) => {
-    userTokenCallback = callback;
-    if (immediate) {
-      callback(values._userToken);
-    }
-  };
-  const viewedObjectIDs = jest.fn();
+export function createInsights() {
+  const analytics = mockSendingEvents(
+    new AlgoliaAnalytics({
+      requestFn: jest.fn(),
+    })
+  );
+  const insightsClient = jest.fn(getFunctionalInterface(analytics));
 
   return {
-    setUserToken,
-    init,
-    _get,
-    onUserTokenChange,
-    viewedObjectIDs,
+    analytics,
+    insightsClient,
   };
 }
 
-export function createInsightsClient(instance = createAlgoliaAnalytics()) {
-  return (methodName, ...args) => {
-    if (!instance[methodName]) {
-      throw new Error(`${methodName} doesn't exist in this mocked instance`);
-    }
-    instance[methodName](...args);
-  };
-}
-
-export function createInsightsUmdVersion(
-  algoliaAnalytics = createAlgoliaAnalytics()
-) {
+export function createInsightsUmdVersion() {
   const globalObject: any = {};
+  globalObject.AlgoliaAnalyticsObject = 'aa';
   globalObject.aa = (...args) => {
     globalObject.aa.queue = globalObject.aa.queue || [];
     globalObject.aa.queue.push(args);
   };
+  const analytics = mockSendingEvents(
+    new AlgoliaAnalytics({
+      requestFn: jest.fn(),
+    })
+  );
 
   return {
+    analytics,
     insightsClient: globalObject.aa,
     libraryLoadedAndProcessQueue: () => {
-      const _aa = createInsightsClient(algoliaAnalytics);
-      const queue = globalObject.aa.queue;
-      queue.forEach(([methodName, ...args]) => {
-        _aa(methodName, ...args);
-      });
-      queue.push = ([methodName, ...args]) => {
-        _aa(methodName, ...args);
-      };
-      return {
-        algoliaAnalytics,
-      };
+      processQueue(analytics, globalObject);
     },
   };
 }
+
+function mockSendingEvents(analytics: AlgoliaAnalytics) {
+  analytics.viewedFilters = jest.fn(analytics.viewedFilters);
+  analytics.viewedObjectIDs = jest.fn(analytics.viewedObjectIDs);
+  analytics.clickedFilters = jest.fn(analytics.clickedFilters);
+  analytics.clickedObjectIDs = jest.fn(analytics.clickedObjectIDs);
+  analytics.clickedObjectIDsAfterSearch = jest.fn(
+    analytics.clickedObjectIDsAfterSearch
+  );
+  analytics.convertedFilters = jest.fn(analytics.convertedFilters);
+  analytics.convertedObjectIDs = jest.fn(analytics.convertedObjectIDs);
+  analytics.convertedObjectIDsAfterSearch = jest.fn(
+    analytics.convertedObjectIDsAfterSearch
+  );
+  return analytics;
+}
+
+function getFunctionalInterface(instance: AlgoliaAnalytics) {
+  return (functionName: string, ...functionArguments: any[]) => {
+    instance[functionName](...functionArguments);
+  };
+}
+
+function processQueue(instance: AlgoliaAnalytics, globalObject: any) {
```

Comment:
In the same spirit of now having to rewrite insights logic here, shouldn't we get some of theses fonctions from the lib ? https://github.com/algolia/search-insights.js/blob/6c58e24abdc8cb7cb334b067a743619c28469db4/lib/_processQueue.ts